src/main/java/io/vertx/ext/couchdb/database/designdoc/DBDesignDoc.java

```diff
@@ -0,0 +1,192 @@
+    this.views.put(viewName, view);
+  }
+
+  /**
+   * Sets the language for this instance.
+   *
+   * @param language the language code to set (e.g., "en", "es", "fr")
+   * @throws IllegalArgumentException if language is null or empty
+   */
+  public void setLanguage(String language) {
+    this.language = language;
+  }
+
+  /**
+   * @return the _id
+   */
+  public String getId() {
+    return _id;
+  }
+
+  /**
+   * Gets the revision identifier of the document.
+   *
+   * @return the revision string that identifies the current version of the document
+   */
+  public String getRev() {
+    return _rev;
+  }
+
+  /**
+   * Converts the design document to a JSON representation compatible with CouchDB format.
+   *
+   * @return A JsonObject containing the design document properties including:
+   *         - _id: The document ID prefixed with "_design/" if not already set
+   *         - _rev: The document revision
+   *         - language: The language used for views
+   *         - views: An object containing all views with their map and reduce functions
+   */
+  public JsonObject toJson() {
+    JsonObject result = new JsonObject();
+    // if new Json need to figure out how to have id and rev
+    if (this.getId() == null || this.getId().isEmpty()) {
+      result.put("_id", "_design/" + this.getName());
+    } else {
+      result.put("_id", this.getId());
+    }
+    result.put("_rev", this.getRev());
+    result.put("language", this.getLanguage());
+    JsonObject viewsObject = new JsonObject();
+    for (Map.Entry<String, DBDesignView> entry : this.views.entrySet()) {
+      // You can directly add any object, Vert.x will handle conversion to JSON types
+      viewsObject.put(entry.getKey(), new JsonObject()
+          .put("map", entry.getValue().getMap())
+          .put("reduce", entry.getValue().getReduce().getValue()));
+    }
+    result.put("views", viewsObject);
+    return result;
+  }
+
+  /**
+   * Converts a JsonObject representation of a design document into a DBDesignDoc object.
+   *
+   * @param dbSecObject The JsonObject containing the design document data. Must contain '_id' and '_rev'
+   *                    fields if isNew is false.
+   * @param isNew       Indicates if this is a new document. If false, '_id' and '_rev' fields are required
+   *                    and validated.
+   * @return A new DBDesignDoc instance populated with data from the JsonObject.
+   * @throws NullPointerException if dbSecObject is null, or if '_id' or '_rev' are null when isNew is false
+   */
+
+  public static DBDesignDoc fromJson(JsonObject dbSecObject, boolean isNew) {
+    if (!isNew) {
+      // id and rev
+      Objects.requireNonNull(dbSecObject.getString("_id"));
+      Objects.requireNonNull(dbSecObject.getString("_rev"));
+    }
+    Objects.requireNonNull(dbSecObject);
+    DBDesignDoc dbDesignDoc = new DBDesignDoc();
+    dbDesignDoc._id = dbSecObject.getString("_id", "");
```

Comment:
since we asserted nonNull `getString("_id")` will do `""~ as default is a code smell